locations/Locations.py

```diff
@@ -0,0 +1,433 @@
+from abc import ABC, abstractmethod
+import re
+
+from utils import get_md5_hash
+
+
+class Location(ABC):
+    """
+    Base location type
+    """
+    def _init_attributes(self):
+        """
+        Default attributes of a location. All fields in attributes should be
+        camelCase since it will be used to generate a JSON resource object.
+        """
+        self.attr = {
+            'name': None,
+            'tags': [],
+            'openHours': {},
+            'type': None,
+            'parent': None,
+            'locationId': None,
+            'abbreviation': None,
+            'geoLocation': None,
+            'geometry': None,
+            'summary': None,
+            'description': None,
+            'descriptionHtml': None,
+            'address': None,
+            'city': None,
+            'state': None,
+            'zip': None,
+            'county': None,
+            'telephone': None,
+            'fax': None,
+            'thumbnails': [],
+            'images': [],
+            'departments': [],
+            'website': None,
+            'sqft': None,
+            'calendar': None,
+            'campus': None,
+            'girCount': None,
+            'girLimit': False,
+            'girLocations': None,
+            'synonyms': [],
+            'bldgId': None,
+            'parkingZoneGroup': None,
+            'propId': None,
+            'adaParkingSpaceCount': None,
+            'motorcycleParkingSpaceCount': None,
+            'evParkingSpaceCount': None,
+            'weeklyMenu': None
+        }
+
+    @abstractmethod
+    def get_primary_id(self):
+        """
+        The function to get location's primary ID and needs to be overrode
+        """
+        pass
```

Comment:
Since these are marked with `@abstractmethod` now, I think you can remove the `pass` statements and just keep the docstring. This seems to be a recommended way of doing it https://stackoverflow.com/a/45826761

Reply:
Good to know, thanks.